From 2e16f47b94a4e227c8f2fbb12f3834ddec5dc035 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 21 Feb 2018 15:08:55 +0000
Subject: [PATCH 6/6] sources: Use bsdtar where possible

Basically normal GNU tar completely flops with overlayfs on my system now
and I just can't be dealing with it. Thus, if we find a bsdtar within the
environment, we're gonna use that guy instead of GNU tar now, to resolve
the "overlayfs eats its face" issue.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/sources.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/ypkg2/sources.py b/ypkg2/sources.py
index 53ee0dc..475a106 100644
--- a/ypkg2/sources.py
+++ b/ypkg2/sources.py
@@ -270,7 +270,12 @@ class TarSource(YpkgSource):
 
     def get_extract_command_tar(self, context, bpath):
         """ Get a command tailored for tar usage """
-        cmd = "tar xf \"{}\" -C \"{}/\"".format(bpath, context.get_build_dir())
+        if os.path.exists("/usr/bin/bsdtar"):
+            frag = "bsdtar"
+        else:
+            frag = "tar"
+        cmd = "{} xf \"{}\" -C \"{}/\"".format(
+            frag, bpath, context.get_build_dir())
         return cmd
 
     def extract(self, context):
-- 
2.16.1

